source/server/http/admin.cc

```diff
@@ -575,13 +574,49 @@ std::string AdminImpl::runtimeAsJson(
   return strbuf.GetString();
 }
 
+Http::Code AdminImpl::handlerHystrixEventStream(const std::string&,
+                                                Http::HeaderMap& response_headers,
+                                                Buffer::Instance&, HandlerInfo& handler_info) {
+
+  response_headers.insertContentType().value().setReference(
+      Http::Headers::get().ContentTypeValues.TextEventStream);
+  response_headers.insertCacheControl().value().setReference(
+      Http::Headers::get().CacheControlValues.NoCache);
+  response_headers.insertConnection().value().setReference(
+      Http::Headers::get().ConnectionValues.Close);
+  response_headers.insertAccessControlAllowHeaders().value().setReference(
+      Http::Headers::get().AccessControlAllowHeadersValue.AccessControlAllowHeadersHystrix);
+  response_headers.insertAccessControlAllowOrigin().value().setReference(
+      Http::Headers::get().AccessControlAllowOriginValue.All);
+  response_headers.insertNoChunks().value().setReference("0");
+
+  Stats::HystrixHandlerInfoImpl& hystrix_handler_info =
+      dynamic_cast<Stats::HystrixHandlerInfoImpl&>(handler_info);
```

Comment:
In fact, the request never terminates. We must use the same socket that was used in the original request for streaming. If the request was terminated, the socket would have been closed. I think this was the main issue with implementing a sink. Will need some help to resolve this.

Reply:
@jmarantz Totally agree with the sentiment.  However, IIUC, the handlers don't have the private variables of the filter available to them since they are methods on the `AdminImpl` object, but I may be missing something.  

Above, we discussed above adding a handler class (rather than just a method) to combine the callback and data and formalize the idea of combining handlers with state.  This would allow handlers to allocate and manage their own state, be tied to the filter object, and not require any `dynamic_cast`ing.  We decided to add this as a TODO rather than do it in this PR, but maybe we should get that in as a part of this PR instead?  What do you think?

Side note: as long as a `bad_cast` exception is the desired result of a failed `dynamic_cast`, I think `dynamic_cast`ing a ref should be fine since it automatically throws if the cast fails.

Reply:
@trabetti I think this will still work exactly the same way as it does now in a sink.  I think the request object would need to be registered with the sink, and then the sink could start encoding data through it just as the timers do now.

Reply:
@mrice32  Good point; how about supplying the AdminFilter* to the callbacks, rather than an empty HandlerInfo* base class that needs to be cast whenever it is used.  That would require passing 'this' to parent_.runCallback() in AdminFilter::onComplete, and then passing that through to the handlers, so they'd have context of the filter instantiation that invoked them.

Reply:
@trabetti IIUC, you're saying you want to break these into two PRs - first the sink that the handler would register with, and then add the handler, itself, in a separate PR?  I think that's a great idea.

@jmarantz yup, seems reasonable to me.

Reply:
@mrice32 I meant for first PR for adding a handler class instead of a method.  Then, modify this PR to use the class.

Reply:
Oh, I see.  Yup, that would be fine too.  I'm always in favor of dividing up PRs where possible :)